Draft : Topology Deployer Web App

vpatel95 opened the pull request.

vpatel95 commented on web-app/ui/src/contexts/UserContext.js:
STALE.

iamprakharjaiswal commented on web-app/ui/src/wrapper/AuthWrapper.js:
```diff
@@ -0,0 +1,12 @@
+
+export const AuthWrapper = () => {
+  const user = SessionStore.getUser();
+
```
yahan pe sessionstore ki jagah auth context use kar skta

vpatel95 replied:
Hn lekin, auth context is not accessible in loader where I need to fetch current user objects using id from the user

iamprakharjaiswal commented on web-app/ui/src/services/api.js:
```diff
@@ -0,0 +1,167 @@
+});
+
+Axios.interceptors.request.use(
+  config => {
```
auth wrapper me check karle if user is authenticated or not, if not call this method and authenticate the user, ideally api methods should only call apis and return the response, they should not set the state

vpatel95 replied:
getToken hai -_-

vpatel95 committed:
Merge pull request #3 from vpatel95/fix-dry-run-flag

Fix dry run flag

vpatel95 committed:
Bump version to 0.1.0

vpatel95 committed:
Recreate networks after host reboot

vpatel95 committed:
Add Config Validations

1. Validate networks on creation
   - Check if interface/bridge exists with the same name
   - Check if provided v4 and v6 subnet collides with existing networks
2. Validate vms on creation
   - Check if vm config has valid fields
   - Check for either flavor or vcpus, disk and ram
   - Check if provided network exists
   - Check if provided IPs belongs to configured subnets
3. Update topologies to version 2
4. Reorganize code and clean up

Closes #18

vpatel95 committed:
Merge pull request #13 from vpatel95/update-help

Add help for options and update README with options

vpatel95 committed:
Update README

vpatel95 committed:
Fix dry run flag

vpatel95 committed:
User cloud image to create VMs.

vpatel95 committed:
Print network IP for VM

vpatel95 committed:
Merge pull request #4 from vpatel95/fix-existing-nw-config

Fix network creation when nw already exists

vpatel95 committed:
Merge pull request #17 from vpatel95/bump-version-0.0.4

Bump version to 0.0.4

vpatel95 committed:
Fix network creation when nw already exists

vpatel95 committed:
Merge pull request #1 from vpatel95/deployer

Create python tool to create topologies

vpatel95 committed:
Update SR-MPLS 6-node topology

vpatel95 committed:
Fix persistent iptables

vpatel95 committed:
Create LICENSE

vpatel95 committed:
Create pre requisite install script

1. Bump version to 0.0.3
2. Added usage in README
3. Clean up deprecated scripts

vpatel95 committed:
Create python-publish.yml

vpatel95 committed:
Add SR-MPLS Ecmp topology

vpatel95 committed:
Pass netmask instead of subnet for mgmt network

vpatel95 committed:
Merge pull request #2 from vpatel95/fix-network-creation

Fix network creation issue

vpatel95 committed:
Bump version to 0.0.4

vpatel95 committed:
Move topology config to topologies folder

vpatel95 committed:
Merge pull request #5 from vpatel95/create-networks

Create networks using deployer

vpatel95 committed:
Clean deployer

vpatel95 committed:
Update SR-MPLS 6-node topology

vpatel95 committed:
Add SR-MPLS Ecmp topology

vpatel95 committed:
Fix network creation issue

vpatel95 committed:
Initial commit

vpatel95 committed:
Add help for options and update README with options

vpatel95 committed:
remove deployer

vpatel95 committed:
Merge pull request #8 from vpatel95/delete-topology

Delete a topology from a given config file

vpatel95 committed:
Initial Commit

vpatel95 committed:
Update README

vpatel95 committed:
Merge pull request #7 from vpatel95/recreate-nw

Recreate networks after host reboot

vpatel95 committed:
Merge pull request #16 from vpatel95/rocky-linux-vms

Add support to bring up Rocky Linux VMs

vpatel95 committed:
Create Nat, Iso and Mgmt network using deployer

1. Depricate the shell scripts.
2. Integrate network creation in the python code
3. Added knobs for avoid creating network/vms and print info

vpatel95 committed:
Create python cli tool

vpatel95 committed:
Add support to bring up Rocky Linux VMs

Closes #15

vpatel95 committed:
Update README with instructions and examples for json config file

vpatel95 committed:
Added Pipfile and Pipfile.lock

vpatel95 committed:
Merge pull request #9 from vpatel95/vpatel95-add-license

Create LICENSE

vpatel95 committed:
fix create_vm script

vpatel95 committed:
Parse topology from JSON config

Added command line arguments
1. --config <config_file> : JSON config
2. --dry-run : List down the commands that will be run without running
   them
3. --log : provide logging level

vpatel95 committed:
Fetch cloud image

vpatel95 committed:
Add create ubuntu dev vm support

vpatel95 committed:
Fix example links in README

vpatel95 committed:
Update setup.py classifier

vpatel95 committed:
Merge pull request #6 from vpatel95/fix-mgmt-nw-netmask

Pass netmask instead of subnet for mgmt network

vpatel95 committed:
Delete a topology from a given config file

vpatel95 committed:
remove mac addr option

vpatel95 committed:
Merge pull request #20 from vpatel95/fix-mgmt-subnet-validation

Check for subnet6 in same config only if configured

vpatel95 committed:
Bump version to 0.2.0

vpatel95 committed:
Merge pull request #28 from vpatel95/bug/custom-vcpu-ram-fix

Fix custom vcpus and ram allocation and unit tests for VM

vpatel95 committed:
Merge pull request #35 from vpatel95/reboot-recovery

Add flag to recover after server reboot

vpatel95 committed:
Merge pull request #25 from vpatel95/expand-ut

Add incremental UTs

vpatel95 committed:
Bump version to 0.1.1

vpatel95 committed:
Added Build Workflow Badge

vpatel95 committed:
Merge pull request #24 from vpatel95/add-workflow-badge

Added Build Workflow Badge

vpatel95 committed:
Bump version to 0.2.1

vpatel95 committed:
Add flag to recover after server reboot

vpatel95 committed:
Added Unit Tests for deployer and network

vpatel95 committed:
Bump version to 0.2.2

vpatel95 committed:
Update TI-LFA topology

vpatel95 committed:
WIP 2 : Initial UI skeleton

vpatel95 committed:
Check for subnet6 in same config only if configured

vpatel95 committed:
Merge pull request #23 from vpatel95/add-tests

Added Unit Tests for deployer and network

vpatel95 committed:
Merge pull request #34 from vpatel95/33-custom-template-path

Provide custom base OS image in config

vpatel95 committed:
Provide custom base OS image in config

vpatel95 committed:
WIP-3 : Fixed database table relations

vpatel95 committed:
Fix custom vcpus and ram allocation and unit tests for VM

vpatel95 committed:
WIP-6 : Add Create forms and fix API bugs

vpatel95 committed:
WIP-5 : Added delete VM handling

vpatel95 committed:
Update README for custom resource value

vpatel95 committed:
Added Sample build custom image script

vpatel95 committed:
Fix flake8 lint errors

vpatel95 committed:
WIP 1 : Api server for topology-deployer

vpatel95 committed:
WIP-4 : Added Network and VM displays

vpatel95 committed:
Merge pull request #19 from vpatel95/validate-config

Config Validation

vpatel95 committed:
Add incremental UTs

vpatel95 committed:
Pass sys.argv to argparse to facilitate tests

vpatel95 committed:
WIP-8 : Create Virtual Machine Form

vpatel95 committed:
WIP-9 : Fix Login page redirection on auth failure

vpatel95 committed:
WIP-7 : State handling, API response format, upgrade bootstrap
